temporal-sdk/src/main/java/io/temporal/client/WorkflowOptions.java

```diff
@@ -281,6 +285,23 @@ public Builder setContextPropagators(@Nullable List<ContextPropagator> contextPr
       return this;
     }
 
+    /**
+     * If {@link WorkflowClient} is used to create a {@link WorkerFactory} that is
+     *
+     * <ul>
+     *   <li>started
+     *   <li>has a non paused worker on the right task queue
+     * </ul>
+     *
+     * TODO
+     *
+     * @param disableEagerExecution if true, an eager local execution of the workflow task will
+     *     never be requested even if it is possible.
+     */
+    public void setDisableEagerExecution(boolean disableEagerExecution) {
```

Comment:
We already have `disableEagerExecution` there. And yeah, with this name... So I think I will just respect this flag for Workflow Tasks too. It looks quite reasonable.
The only problem is that it's a bit backward from implementation perspective, because this flag is on Worker, but used by a Client. But nevertheless I think WorkerOptions is the right place for it from a user intuition perspective.

Reply:
I'm not sure I get the scenario where a client would want to set this flag?

Reply:
I think it's the general safety of being able to turn off this new unbalancing feature. Granted I think it may need to be disable-able at a higher level, but I could _technically_ see doing it for cert workflows.

One use case I can see is that we've been telling people they can use poll success/timeout metrics to determine they have too many workers and can use that to scale down. Well, that's gonna completely change now for people with colocated clients and workers. Especially in cases where the client calls are not evenly distributed, workers without those client calls will all the sudden start hitting poll timeouts more because work distribution isn't as uniform as it once was (not that it was that uniform/fair in the first place though).

Reply:
I can see a case where a worker can process latency sensitive and insensitive workflows and you'd want to disable eager workflows for the latency insensitive ones.